colab-chat/colab-server
# Replace Redis with Kafka

Fixes #11 

- [x] Merge #23 first, I'll have to then fix the formatting problems which flake8 finds before we merge this PR.

NB, dependencies have changed so you'll want to `pip install -r requirements`, but you'll first need to download and install `librdkafka` version 0.9.5 by doing `sudo bash travis.sh`.

## To test
- Start with `docker-compose up --build --remove-orphans`, "--remove-orphans" will get rid of the redis container which is no longer needed.
- Wait until you see the web container is up *
- Go to `localhost` in your browser, login etc
- Go to `localhost/hello` in another browser window/tab, this will send a message "Life is good!" which should be received via server-sent events in the first window.

*you may notice that the web app is slow to start. This is because I have put in a delay to ensure that Kafka is ready first. A more satisfactory fix would be to add a custom "healthcheck" in docker compose so that the web container is started as soon as Kafka is ready. I've created a ticket here: #19
